### PR TITLE
Nightmode Fixes

### PR DESCRIPTION
**Pull request by Lohikar**
changes: 
  - rscadd: "Added a night-mode control program for the Chief Engineer."
  - tweak: "Changed how night-mode works internally."
  - tweak: "Red alert now disables night-mode."
  - spellcheck: "Slightly changed alert messages."
  - bugfix: "Speculative fix for perpetual night-mode."

### CODE REVIEW

File: html/changelogs/Lohikar-NightmodeTweaks.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rscadd: "Added a night-mode control program for the Chief Engineer."
+  - tweak: "Changed how night-mode works internally."
+  - tweak: "Red alert now disables night-mode."
```
Comment (NanakoAC) [inline html/changelogs/Lohikar-NightmodeTweaks.yml]: It's good to see some technical work on it, although bedshaped was going to fix it anyways. I'm not sure it needed such a drastic overhaul

I'm not keen on red alert disabling it. thats a game balance thing, night mode can be very helpful to all kinds of antag activities, which are also the thing that causes red alert. Let people turn it off individually in that situation if ordered

I don't like restricting where it happens either. Night mode in the hallways, departures, etc has proven to be far more interesting and immersive. I don't see a reason to restrict it from anywhere, if someone needs more light in their lab they can toggle it manually.

It's a no from me on the design decisions here, although ill leave it to bedshaped to address the technical specifics

Comment (Lohikar) [inline html/changelogs/Lohikar-NightmodeTweaks.yml]: The actual places affected by night-mode should be unchanged from the original code, the no_light_control flag was/is for disabling lights and doesn't affect night-mode.

Comment (skull132) [inline html/changelogs/Lohikar-NightmodeTweaks.yml]: If you look at the list of areas originally affected by night lighting, Nanako, then you'll see that there's no change real change.

Only lobbies and hallways, plus the crew quarters, were affected. Which is a reasonable choice from the standpoint of believability.

**Comment by skull132** [comment]
Fixes #1200

File: html/changelogs/Lohikar-NightmodeTweaks.yml (hunk 1 above)
Comment (NanakoAC) [inline html/changelogs/Lohikar-NightmodeTweaks.yml]: As its his work, i'd still really like to see bedshaped sign off on this, i'm uncertain about it atm

**Comment by Bedshaped** [comment]
I'm not sure if decentralising the whitelist is such a good idea. Leaving yourself open to potential runtime errors and null pointers. I'm very happy a feature I worked on is being modified and expanded by someone else though 🙂

**Comment by skull132** [comment]
Null pointers are impossible, as the var is checked at compile time. The only thing you may have happen is that someone touches the area define and you don't notice. But looking at the night lighting controller you can figure it out relatively quickly.

One thing: medical should be added to the blacklist. Because surgery in the dark is cancer.

File: code/controllers/Processes/night_lighting.dm [hunk 2]
```diff
@@ -60,26 +44,28 @@
...
+/datum/controller/process/night_lighting/proc/deactivate(var/whitelisted_only = 1)
+	isactive = 0
+	for (var/obj/machinery/power/apc/APC in get_apc_list(whitelisted_only))
 		APC.toggle_nightlight("off")
-		isactive = 0
 
 		SCHECK
 
-/datum/controller/process/night_lighting/proc/get_apc_list()
+/datum/controller/process/night_lighting/proc/get_apc_list(var/whitelisted_only = 1)
```
Comment (skull132) [inline code/controllers/Processes/night_lighting.dm]: We was thinking. Cutting the AI control wire (which also technically means remote control in general) should make the console also no longer affect said APCs. Consistency's sake, primarily.

```
if (B.apc && !B.apc.aidisabled)
	lighting_apcs += B.apc
```
Changing line 70 condition to that would cover it.

**Comment by skull132** [comment]
Bedshaped approved it over Discord. Awaaaay, we go.

**Pull request merged**